src/realm/sync/client.cpp

```diff
@@ -1607,7 +1607,7 @@ ClientImpl::Connection::Connection(ClientImpl& client, connection_ident_type ide
             // Connection object may be destroyed now.
         }
     };
-    m_on_idle = network::Trigger{client.get_service(), std::move(handler)}; // Throws
+    m_on_idle = std::make_shared<Trigger>(&client.get_service(), std::move(handler)); // Throws
```

Comment:
Let's move the handler definition into the Trigger constructor and update the handlers to take a Status parameter, like this:
```
m_on_idle = std::make_shared<Trigger>(&client.get_service(), [this](Status status) {
        if (!status.is_ok())
            return;
        REALM_ASSERT(m_activated);
        if (m_state == ConnectionState::disconnected && m_num_active_sessions == 0) {
            on_idle(); // Throws
            // Connection object may be destroyed now.
        }
    });
```